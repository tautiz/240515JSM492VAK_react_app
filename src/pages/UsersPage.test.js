import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import UserPage from './UsersPage';

describe('UserPage Component', () => {
  const renderWithRouter = (userId) => {
    return render(
      <MemoryRouter initialEntries={[`/user/${userId}`]}>
        <Routes>
          <Route path="/user/:id" element={<UserPage />} />
        </Routes>
      </MemoryRouter>
    );
  };

  test('renders user page title', () => {
    renderWithRouter('123');
    const titleElement = screen.getByText(/User Page/i);
    expect(titleElement).toBeInTheDocument();
  });

  test('displays correct user ID from URL params', () => {
    const testUserId = '123';
    renderWithRouter(testUserId);
    const userIdText = screen.getByText(new RegExp(testUserId));
    expect(userIdText).toBeInTheDocument();
  });

  test('displays user information text', () => {
    renderWithRouter('123');
    const infoText = screen.getByText(/Rodymas informacijos apie vartotojÄ… su ID/i);
    expect(infoText).toBeInTheDocument();
  });

  test('handles different user IDs', () => {
    const testUserId = '456';
    renderWithRouter(testUserId);
    const userIdText = screen.getByText(new RegExp(testUserId));
    expect(userIdText).toBeInTheDocument();
  });

  test('component structure matches snapshot', () => {
    const { container } = renderWithRouter('123');
    expect(container.firstChild).toMatchSnapshot();
  });
});
