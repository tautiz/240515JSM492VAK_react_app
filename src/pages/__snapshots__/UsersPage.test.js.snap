// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPage Component component structure matches snapshot 1`] = `
<div>
  <h2>
    User Page
  </h2>
  <p>
    Rodymas informacijos apie vartotojÄ… su ID: 
    123
  </p>
</div>
`;
